sdks/python/apache_beam/runners/worker/sdk_worker_main.py

```diff
@@ -197,6 +198,19 @@ def _get_worker_count(pipeline_options):
   return 12
 
 
+def _load_avro_coder(pipeline_options):
+  experiments = pipeline_options.view_as(DebugOptions).experiments
+
+  experiments = experiments if experiments else []
+
+  for experiment in experiments:
+    # There should only be 1 match so returning from the loop
+    if re.match(r'xlang_test', experiment):
```

Comment:
I think having to do logic specific to a test in sdk_worker_main.py is pretty hacky.

Reply:
same here.